sys-devel/binutils/binutils-2.31.1.recipe

```diff
@@ -0,0 +1,192 @@
+SUMMARY="Assembler, linker and binary tools for target ${targetMachineTriple}"
+DESCRIPTION="The GNU Binutils are a collection of binary tools.
+
+The main ones are:
+- ld - the GNU linker.
+- as - the GNU assembler.
+
+But they also include:
+- addr2line - converts addresses into filenames and line numbers.
+- ar - a utility for creating, modifying and extracting of archives.
+- c++filt - filter to demangle encoded C++ symbols.
+- gold - a linker for ELF files.
+- nm - lists symbols from object files.
+- objcopy - copys and translates object files.
+- objdump - displays information from object files.
+- ranlib - generates an index to the contents of an archive.
+- readelf - displays information from any ELF format object file.
+- size - lists the section sizes of an object or archive file.
+- strings - lists printable strings from files.
+- strip - discards symbols.
+
+The tools of this package can be used for cross-builds to \
+$effectiveTargetMachineTriple."
+HOMEPAGE="https://www.gnu.org/software/binutils/"
+COPYRIGHT="1988-2018 Free Software Foundation, Inc."
+LICENSE="GNU GPL v3
+	GNU LGPL v3"
+REVISION="1"
+binutilsVersion="2.31.1"
+SOURCE_URI="https://ftpmirror.gnu.org/binutils/binutils-$binutilsVersion.tar.xz
+	https://ftp.gnu.org/gnu/binutils/binutils-$binutilsVersion.tar.xz"
+CHECKSUM_SHA256="5d20086ecf5752cc7d9134246e9588fa201740d540f7eb84d795b1f7a93bca86"
+SOURCE_DIR="binutils-$binutilsVersion"
+PATCHES="binutils-$portVersion.patchset"
+
+# disabled on x86: see https://github.com/haikuports/haikuports/issues/1750
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64 ?arm"
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	binutils$secondaryArchSuffix = $portVersion compat >= 2.23
+	cmd:addr2line$secondaryArchSuffix = $portVersion compat >= 2.23
+	cmd:ar$secondaryArchSuffix = $portVersion compat >= 2.23
+	cmd:as$secondaryArchSuffix = $portVersion compat >= 2.23
+	cmd:c++filt$secondaryArchSuffix = $portVersion compat >= 2.23
+	cmd:elfedit$secondaryArchSuffix = $portVersion compat >= 2.23
+	cmd:gprof$secondaryArchSuffix = $portVersion compat >= 2.23
+	cmd:ld$secondaryArchSuffix = $portVersion compat >= 2.23
+	cmd:ld.bfd$secondaryArchSuffix = $portVersion compat >= 2.23
```

Comment:
I found it weird that `--enable-gold` were passed but not a part of the package (`ld.gold`). Can you check?